openedx/core/djangoapps/discussions/api/external_discussions.py

```diff
@@ -0,0 +1,90 @@
+    """
+
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .create(
+                context_key=context_key,
+                usage_key=usage_key,
+                external_discussion_id=external_discussion_id
+            )
+    except IntegrityError:
+        raise ValueError(
+            "Unable to create External Discussion Mapping. A mapping already exists for "
+            "context_key={context_key} and usage_key={usage_key}".format(
+                context_key=context_key,
+                usage_key=usage_key
+            )
+        )
+
+    return ExternalDiscussionData(
+        context_key=discussion_mapping.context_key,
+        external_discussion_id=discussion_mapping.external_discussion_id,
+        usage_key=discussion_mapping.usage_key
+    )
+
+
+def remove_external_discussion_mapping(external_discussion_data: ExternalDiscussionData) -> bool:
+    """
+    Given an ExternalDiscussionData object, attempt to remove the corresponding DB object.
+    Returns True on success, and False on failure
+    """
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(
+                context_key=external_discussion_data.context_key,
+                external_discussion_id=external_discussion_data.external_discussion_id,
+                usage_key=external_discussion_data.usage_key
+            )
+        discussion_mapping.delete()
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        return False
+
+    return True
+
+
+def get_external_discussion_context(context_key: LearningContextKey, usage_key: UsageKey) -> ExternalDiscussionData:
+    """
+    Return an ExternalDiscussionData object with the external_discussion_id given the context_key and usage_key
+    """
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(context_key=context_key, usage_key=usage_key)
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        return None
+
+    return ExternalDiscussionData(
+        context_key=discussion_mapping.context_key,
+        external_discussion_id=discussion_mapping.external_discussion_id,
+        usage_key=discussion_mapping.usage_key
+    )
+
+
+def get_internal_discussion_context(external_discussion_id: str) -> ExternalDiscussionData:
+    """
+    Return an ExternalDiscussionData object with the context_key and usage_key given the external_discussion_id
+    """
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(external_discussion_id=external_discussion_id)
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        return None
+
+    return ExternalDiscussionData(
+        context_key=discussion_mapping.context_key,
+        external_discussion_id=discussion_mapping.external_discussion_id,
+        usage_key=discussion_mapping.usage_key
+    )
```

Comment:
I just realised that we can't really guarantee that the external discussion id will be unique, for a single provider or across providers. 

What I mean by this is, a discussion provider might have a slug external id like "introductions". When combined with the providers credentials this might be unique, but this slug might exist for multiple discussion providers, across multiple instances. 

So, we can't be sure a single item will be returned here and, more importantly, the multiple items returned could be across courses, and providers and provider configurations. 

I think we should just remove this API for now and look into it if the use case for this ever comes up.

Reply:
Sounds good. Removing this and relevant tests.